test/python/circuit/test_parameters.py

```diff
@@ -1178,7 +1178,9 @@ def test_compare_to_value_when_bound(self):
 
         x = Parameter("x")
         bound_expr = x.bind({x: 2.3})
+
         self.assertEqual(bound_expr, 2.3)
+        self.assertEqual(abs(bound_expr), 2.3)
```

Comment:
I think it would be best to separate out the tests of `abs` into their own function; they're not really related to this.

Also, I think we could do with more stringent tests of it; at the moment, if `abs` is implemented incorrectly this test can still fail.  We need to actually test the mathematical properties of `abs`, for example that it makes negative numbers positive, and complex numbers positive-real.  We should also test that it works for symbolic expressions, such as `abs(< -x >)`.

Reply:
As suggested, I added more tests to the implementation. What do you mean by testing symbolic expressions, such as `abs(< -x >)`?

While doing this I also encountered an interesting problem with ```__eq```:
```
x = Parameter("x")
y = x + 1 - 1
assert(x, y) # return False
```
The solution is to also use ```NotImplement``` as return. Do you want to fix that separately?